Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -370,26 +541,23 @@ void CDMInstanceOpenCDM::loadSession(LicenseType, const String& sessionId, const
         GST_WARNING("cannot load the session %s cause we can't find it", sessionId.utf8().data());
         return;
     }
-    SessionLoadFailure sessionFailure = SessionLoadFailure::None;
-
-    auto result = session->load();
-    if (result.first) {
-        if (!result.second.startsWith("message:")) {
-            SharedBuffer& initData(session->initData());
-            KeyStatusVector knownKeys;
-            MediaKeyStatus keyStatus = mediaKeyStatusFromOpenCDM(result.second);
-            knownKeys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus>{initData, keyStatus});
-            GST_DEBUG("session %s loaded, status %s", sessionId.utf8().data(), result.second.utf8().data());
-            callback(WTFMove(knownKeys), std::nullopt, std::nullopt, SuccessValue::Succeeded, sessionFailure);
+    session->load([callback = WTFMove(callback)](Session* session, bool success, const String& response) {
+        if (success) {
+            if (!response.startsWith("message:")) {
+                MediaKeyStatus keyStatus = mediaKeyStatusFromOpenCDM(response);
+                KeyStatusVector knownKeys;
+                knownKeys.append(std::pair<Ref<SharedBuffer>, MediaKeyStatus>{session->initData()->copy(), keyStatus});
```

Comment:
You can pass `mediaKeyStatusFromOpenCDM(response)` instead of `keyStatus` and avoid that variable